### ISSUE TITLE
Update JS API Spec for v5

### ISSUE THREAD

**Issue by sergeychernyshev** [ux-capture/ux-capture#117]
Make updates to [core library JS API Spec](https://github.com/ux-capture/ux-capture/blob/main/packages/ux-capture/docs/ux-capture-js-api-spec.md) to reflect any changes.

These are the ones that immediately come to mind:
- [ ] Library can now be moved out of the critical path as it will use `buffered` observers
- [ ] Expected event definition will no longer need to happen before UI is rendered
- [ ] Some lightweight wrapper might still need to stay inlined at the top of the page (if "wait till next frame" trick is still needed)
- [ ] New sequence diagram to reflect it all
- [ ] Native Soft Navigation support and fallback scenarios will have to be spelled out (it is a candidate for subsequent versions and might have to be descoped from initial version)
